### PR TITLE
[DRAFT] Add token api

### PR DESCRIPTION
**Pull request by areshand**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

(You must have submitted a [signed CLA](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#contributor-license-agreement) that includes your GitHub handle prior to us accepting and landing your work. Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

### CODE REVIEW

File: api/doc/openapi.yaml [hunk 1]
```diff
@@ -504,6 +504,147 @@ paths:
...
           $ref: '#/components/responses/500'
+  /tokens/{TokenId}:
+    get:
+      description: "Returns Token by Id"
+      operationId: getTokenById
+      parameters:
+        - name: "TokenId"
+          in: path
+          required: true
+          description: "Token Id, has format `${address}:${collection_name}:${token_name}`"
```
Comment (davidiw) [inline api/doc/openapi.yaml:515]: creators_address

File: api/doc/openapi.yaml [hunk 2]
```diff
@@ -504,6 +504,147 @@ paths:
...
+        '404':
+          $ref: '#/components/responses/404'
+        '400':
+          $ref: '#/components/responses/400'
+        '500':
+          $ref: '#/components/responses/500'
+  /token_metadata/{tokenId}:
+    get:
+      description: "Returns TokenMetaData by Id"
+      operationId: getTokenMetaDataById
```
Comment (davidiw) [inline api/doc/openapi.yaml:534]: Metadata and only MetaData (in addition) if there's an issue with another library

**Review by davidiw — COMMENTED** [review]
well thanks to reviewing this, I have learned more about "proper" Rest: https://restfulapi.net/resource-naming/

this looks pretty good

File: api/doc/openapi.yaml (hunk 1 above)
Comment (areshand) [reply api/doc/openapi.yaml:515]: uh, the actual PR is here https://github.com/aptos-labs/aptos-core/pull/1113 . will update that based on the comment here.

**Pull request closed without merging**